cea/optimization/preprocessing/preprocessing_main.py

```diff
@@ -129,23 +131,19 @@ def __init__(self, locator, building_names):
 #============================
 
 
-def run_as_script(scenario_path=None):
+def main(config):
     """
     run the whole preprocessing routine
     """
-    import cea.config
-    config = cea.config.Configuration()
     gv = cea.globalvar.GlobalVariables()
-
-
-    locator = cea.inputlocator.InputLocator(scenario_path=config.scenario)
+    locator = cea.inputlocator.InputLocator(scenario=config.scenario)
     total_demand = pd.read_csv(locator.get_total_demand())
     building_names = pd.read_csv(locator.get_total_demand())['Name']
```

Comment:
this happens a lot in CEA code and @martin-mosteiro  and I just found out that we need a unified way of getting the names of the buildings in a scenario. There's an open issue and you are gladly invited to come up with a nice solution to share :-)

Reply:
(see issue #934)